src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java

```diff
@@ -95,75 +96,97 @@ public void setupMaps() {
 		 * - apps can start and stop, we need to see this, too
 		 * - instances can be added to apps
 		 * - Blue/green deployment may alter both of them
-		 * 
+		 *
 		 * In short: both are very volatile and we need to query them often
 		 */
 		this.applicationMap = new PassiveExpiringMap<>(this.timeoutCacheApplicationLevel, TimeUnit.SECONDS);
 		this.hostnameMap = new PassiveExpiringMap<>(this.timeoutCacheApplicationLevel, TimeUnit.SECONDS);
 		this.domainMap = new PassiveExpiringMap<>(this.timeoutCacheApplicationLevel, TimeUnit.SECONDS);
 	}
-	
-	
+
+
 	public List<Instance> determineInstancesFromTargets(List<Target> targets) {
 		LinkedList<Instance> result = new LinkedList<>();
-		
-		Flux<Target> initialFlux = Flux.fromIterable(targets);
-		
-		Flux<InternalInstance> initialInstancesOnlyApplication = initialFlux.map( target -> {
+
+		Flux<Target> initialFlux = this.expandTargets(Flux.fromIterable(targets));
```

Comment:
I am unsure if this is a great idea or not (perhaps you may comment on this from your point of view): Expanding the targets will be done on every call of `determineInstancesFromTargets`, which means on every request to scrape. As I assume that a pre-selection request will be fired (see also other code review comment), this would mean that this wildcard resolution bypasses the cache. 
Whilst this is a good idea from a consistency's point of view, it will be negative from the performance's / TCO's perspective. 

Was this decision pro consistency intentional?

Reply:
I believe this overlaps with the other comments above, e.g. finding a efficient and less intrusive mechanisms to invalidate a and refresh the metadata?